src/detect-engine-event.c

```diff
@@ -56,6 +58,70 @@ static void DetectEngineEventFree (DetectEngineCtx *, void *);
 void EngineEventRegisterTests(void);
 #endif
 
+static bool PrefilterEventIsPrefilterable(const Signature *s, int smtype)
+{
+    const SigMatch *sm;
+    for (sm = s->init_data->smlists[DETECT_SM_LIST_MATCH]; sm != NULL; sm = sm->next) {
+        switch (sm->type) {
+            case DETECT_ENGINE_EVENT:
```

Comment:
I think this should be `smtype` as the fn is reused for Engine as well as Decode events. wdyt?

Reply:
right, will fix this